### PR TITLE
[DOCFIX] Update Code-Convention Page with Javadoc guide

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

File: docs/en/contributor/Code-Conventions.md [hunk 1]
```diff
@@ -75,6 +75,57 @@ warnings are introduced:
...
+ * @param param_n description of nth parameter
+ * @return description of return argument (if applicable)
+ * @throws exception_1 description of 1st exception case
+ * ...
+ * @throws exception_n description of nth exception case
+ */
+```
+- An exception to the above rule is that `@throws` doesn’t need to be provided for `@Test` methods,
+or for generic exceptions like IOException when there is nothing interesting to document.
+- Do not put unchecked exceptions like `RuntimeException` in javadoc unless it is critical for this methd.
```
Comment (gpang) [inline docs/en/contributor/Code-Conventions.md:110]: We may want to mention something like this (https://docs.alluxio.io/os/user/stable/en/contributor/Code-Conventions.html#selectively-javadoc-exceptions) here as well.

**Comment by gpang** [comment]
From here (https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#descriptions) there are also guidelines for writing the description. Some we can make explicit here are:

- first sentence should be a concise summary of the class or method.
- the description should generally be implementation-independent.

We might also want descriptions (not the first sentence) to include any significant performance implications.

File: docs/en/contributor/Code-Conventions.md (hunk 1 above)
Comment (apc999) [reply docs/en/contributor/Code-Conventions.md:110]: done

**Comment by apc999** [comment]
@gpang  updated , Ptal

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by apc999** [comment]
alluxio-bot, cherry-pick this to branch-2.3 please

**Pull request merged**